apps/laboratory/src/components/Solana/SolanaSendTransactionTest.tsx

```diff
@@ -0,0 +1,163 @@
+      }
+
+      // Create a new transaction
+      const transaction = new Transaction().add(
+        SystemProgram.transfer({
+          fromPubkey: walletProvider.publicKey,
+          toPubkey: recipientAddress,
+          lamports: amountInLamports
+        })
+      )
+      transaction.feePayer = walletProvider.publicKey
+
+      const { blockhash } = await connection.getLatestBlockhash()
+
+      transaction.recentBlockhash = blockhash
+
+      const signature = await walletProvider.sendTransaction(transaction, connection as Connection)
+      toast({ title: 'Succcess', description: signature, status: 'success', isClosable: true })
+    } catch (err) {
+      toast({
+        title: 'Error',
+        description: (err as Error).message,
+        status: 'error',
+        isClosable: true
+      })
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  async function onSendVersionedTransaction() {
+    try {
+      setLoading(true)
+      if (!walletProvider || !address) {
+        throw Error('user is disconnected')
+      }
+
+      if (!connection) {
+        throw Error('no connection set')
+      }
+
+      const balance = await connection.getBalance(walletProvider.publicKey)
+      if (balance < amountInLamports) {
+        throw Error('Not enough SOL in wallet')
+      }
+
+      const { blockhash } = await connection.getLatestBlockhash()
+
+      const instructions = [
+        SystemProgram.transfer({
+          fromPubkey: walletProvider.publicKey,
+          toPubkey: recipientAddress,
+          lamports: amountInLamports
+        })
+      ]
+
+      // Create v0 compatible message
+      const messageV0 = new TransactionMessage({
+        payerKey: walletProvider.publicKey,
+        recentBlockhash: blockhash,
+        instructions
+      }).compileToV0Message()
+
+      // Make a versioned transaction
+      const transactionV0 = new VersionedTransaction(messageV0)
+
+      const signature = await walletProvider.sendTransaction(
+        transactionV0,
+        connection as Connection
+      )
+      toast({ title: 'Succcess', description: signature, status: 'success', isClosable: true })
+    } catch (err) {
+      toast({
+        title: 'Error',
+        description: (err as Error).message,
+        status: 'error',
+        isClosable: true
+      })
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  if (!address) {
+    return null
+  }
+
+  if (chainId === solana.chainId) {
+    return (
+      <Text fontSize="md" color="yellow">
+        Switch to Solana Devnet or Testnet to test this feature
+      </Text>
+    )
+  }
+
+  const supportV0Transactions = walletProvider.name !== 'WalletConnect'
+
+  return (
+    <Stack direction={['column', 'column', 'row']}>
+      <Button
+        data-test-id="sign-transaction-button"
+        onClick={onSendTransaction}
+        isDisabled={loading}
+      >
+        Sign and Send Transaction
+      </Button>
+      {supportV0Transactions && (
```

Comment:
While this will work as expected, let's stick to using ternary operator as I explained in the previous comment